common/lib/xmodule/xmodule/modulestore/tests/test_semantics.py

```diff
@@ -0,0 +1,257 @@
+    @ddt.unpack
+    def test_delete_child(self, block_type, child_published):
+        block_usage_key = self.course.id.make_usage_key(block_type, 'test_block')
+        child_usage_key = self.course.id.make_usage_key('html', 'test_child')
+
+        self.assertCourseDoesntPointToBlock(block_usage_key)
+        self.assertBlockDoesntExist(block_usage_key)
+        self.assertBlockDoesntExist(child_usage_key)
+
+        test_data = self.DATA_FIELDS[block_type]
+        child_data = '<div>child value</div>'
+
+        with self.store.branch_setting(ModuleStoreEnum.Branch.draft_preferred):
+            block = self.store.create_child(
+                user_id=ModuleStoreEnum.UserID.test,
+                parent_usage_key=self.course.scope_ids.usage_id,
+                block_type=block_type,
+                block_id=block_usage_key.block_id,
+                fields = {test_data.field_name: test_data.initial},
+            )
+
+            child = self.store.create_child(
+                user_id=ModuleStoreEnum.UserID.test,
+                parent_usage_key=block_usage_key,
+                block_type=child_usage_key.block_type,
+                block_id=child_usage_key.block_id,
+                fields = {'data': child_data},
+            )
+
+        if child_published:
+            self.store.publish(child_usage_key, ModuleStoreEnum.UserID.test)
+
+        self.assertCoursePointsToBlock(block_usage_key)
+
+        if child_published:
+            self.assertParentOf(block_usage_key, child_usage_key)
+        else:
+            self.assertParentOf(block_usage_key, child_usage_key, draft=True)
+            # N.B. whether the direct-only parent block points to the child in the publish branch
+            # is left as undefined behavior
+
+        self.assertBlockHasContent(block_usage_key, test_data.field_name, test_data.initial)
+
+        if child_published:
+            self.assertBlockHasContent(child_usage_key, 'data', child_data)
+        else:
+            self.assertBlockHasContent(child_usage_key, 'data', child_data, draft=True)
+            self.assertBlockDoesntExist(child_usage_key, draft=False)
+
+        with self.store.branch_setting(ModuleStoreEnum.Branch.draft_preferred):
+            self.store.delete_item(child_usage_key, ModuleStoreEnum.UserID.test)
+
+        self.assertCoursePointsToBlock(block_usage_key)
+        self.assertNotParentOf(block_usage_key, child_usage_key)
+
+        if child_published and self.store.get_modulestore_type(self.course.id) == ModuleStoreEnum.Type.mongo:
+            # N.B. This block is being left as an orphan in old-mongo. This test will
+            # fail when that is fixed. At that time, this condition should just be removed,
+            # as SplitMongo and OldMongo will have the same semantics.
+            self.assertBlockHasContent(child_usage_key, 'data', child_data)
+        else:
+            self.assertBlockDoesntExist(child_usage_key)
```

Comment:
Sigh. It's because the course wasn't actually being built in mongo, it was being built in split.